packages/yoastseo/src/languageProcessing/researches/keywordCountInUrl.js

```diff
@@ -103,7 +12,20 @@ function dehyphenateKeyphraseForms( topicForms ) {
  * @returns {{keyphraseLength: int, percentWordMatches: int}} The length of the keyphrase and the percentage of keyphrase forms that has been found.
  */
 function keywordCountInSlug( paper, researcher ) {
-	const topicForms = dehyphenateKeyphraseForms( researcher.getResearch( "morphology" ) );
+	/*
+	 * In most languages we treat hyphens as word boundaries, but there are exceptions, such as Indonesian.
+	 * However, for matching the keyphrase in the slug we want to split the keyphrase forms on hyphens in all languages.
+	 * This is because slugs usually use hyphens (or underscores) as word boundaries. So if only the slug is split on
+	 * hyphens and not the keyphrase forms, some keyphrase forms won't be found in the slug.
+	 * That's why if a researcher's config states that hyphens *shouldn't* be treated as word boundaries, we run the
+	 * morphology research not from the researcher but instead directly from this file, and pass an extra flag that
+	 * tells it to ignore the areHyphensWordBoundaries config from the researcher. This way we treat hyphens as word
+	 * boundaries in all languages when retrieving the keyphrase and synonym forms to be used for finding the keyphrase
+	 * in slug.
+	 */
+	const topicForms = researcher.getConfig( "areHyphensWordBoundaries" ) ? researcher.getResearch( "morphology" )
+		: getWordForms( paper, researcher, true );
```

Comment:
```suggestion
	const languageResearcher = cloneDeep( researcher );
	languageResearcher.addConfig( "areHyphensWordBoundaries", true );
	const topicForms = languageResearcher.getResearch( "morphology" );
```

Reply:
> ⚠️ It's important to clone the researcher before overriding the value of `areHyphensWordBoundaries`. This is to avoid overriding its value everywhere else the researcher is called.
> ⚠️ In order to be able to use `.addConfig()` with a boolean parameter, you also need to make a small change in `AbstractResearcher` 's `addConfig()` method to not throw an error when we add a boolean value as a config. `isEmpty()` recognizes boolean `true` and `false` as empty. So there you need to add an extra check to throw the error when the config is empty and is not a primitive data:

````
addConfig( name, config ) {
...
		if ( isUndefined( config ) || ( isEmpty( config ) && config === Object( config ) ) ) {
			throw new MissingArgument( "Failed to add the custom researcher config. Config cannot be empty." );
		}
...
````